cli/flox/src/commands/build.rs

```diff
@@ -302,6 +305,77 @@ pub(crate) fn check_stability_compatibility<'p>(
     Ok(())
 }
 
+/// Enforce the existence of a git repository when building nix expressions,
+/// to avoid costly and potentially insecure copies to the nix store.
+/// Additionally, ensure that the expression files are tracked by git,
+/// so that they are guaranteed to be found by the build subsystem,
+/// which filters any untracked sources
+/// allowing us to provide cleaner messaging on the way.
+pub(crate) fn check_git_tracking_for_expression_builds<'p>(
+    packages_to_build: impl IntoIterator<Item = &'p PackageTarget>,
+    expression_dir: &Path,
+) -> Result<()> {
+    let mut expression_builds = packages_to_build
+        .into_iter()
+        .filter(|target| target.kind().is_expression_build())
+        .peekable();
+
+    if expression_builds.peek().is_none() {
+        return Ok(());
+    }
+
+    let expression_builds: Vec<_> = expression_builds
+        .map(|target| {
+            let PackageTargetKind::ExpressionBuild(metadata) = target.kind() else {
+                unreachable!("kind checked above");
+            };
+            (target.name(), metadata)
+        })
+        .collect();
+
+    let expression_builds_formatted = expression_builds
+        .iter()
+        .map(|(name, _)| format!("  - {name}"))
+        .join("\n");
+
+    let git = match GitCommandProvider::discover(expression_dir) {
+        Err(err) => {
+            trace!(%err, "git discovery error");
+
+            bail!(formatdoc! {"
+                Building nix expression build(s) requires git version control.
+                Only git tracked files (including the expressions themselves)
+                will be available during nix expression builds.
```

Comment:
Still think this (and the error below) should be formatted slightly different to match the [style guide](https://www.notion.so/floxdev/Flox-CLI-Style-Guide-ff785fd9bdbc445284ab6e1c660984a8?source=copy_link#15dfa908387647959f3e6730873cd35e).
```suggestion
                Building nix expression build(s) requires git version control.
                
                Only git tracked files (including the expressions themselves) will be available during nix expression builds.
```